lib/common/version_handler.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module ReleaseManager
+  module Common
+    class VersionHandler
+      def change_version(revision, release_type)
+        if !release_type || release_type == 'z'
+          change_version_z(revision)
+        else
+          change_version_y(revision)
+        end
+      end
+
+      def add_versions(name, component, release_type)
+        return component if should_add_versions?(name)
+
+        component[:current_version] = extract_revision(component[:tag])
+        component[:suggested] = calculate_next(component, release_type)
+        component
+      end
+
+      def should_add_versions?(name)
+        name.match?(/core|runtime|api/)
+      end
+
+      private
+
+      def extract_revision(tag)
+        tag.match(/[0-9]+\.[0-9]+\.[0-9]+/).to_s
+      end
+
+      def calculate_next(component, release_type)
+        if component[:commits].any?
```

Comment:
This can be done as a 2-liner if you use return. Eg. `return component[:current_version] unless component[:commits].any?`